conans/client/rest/conan_requester.py

```diff
@@ -12,7 +13,14 @@
 class ConanRequester(object):
 
     def __init__(self, config, http_requester=None):
-        self._http_requester = http_requester if http_requester else requests.Session()
+        if http_requester:
+            self._http_requester = http_requester
+        else:
+            self._http_requester = requests.Session()
+            adapter = HTTPAdapter(max_retries=config.retry)
```

Comment:
Not sure if this should be associated with `config.retry` or not. This retry is about connection retry, right? Should we create a new config for ìt? `retry_connection`? @memsharded

Reply:
yes, about connection

Reply:
This sounds a bit extreme. So far there hasn't been any other report that indicate that this is needed, and with this change it seems that now absolutely every connection will retry the specified number of times (the default is None), because of a edge case on the Azure platform. 

We need to make sure that this is not affecting anything else in other ways. Maybe we want to restrict the retries to some conditions?

```python
s = requests.Session()
retries = Retry(total=5, backoff_factor=1, status_forcelist=[ ... ])
s.mount('http://', HTTPAdapter(max_retries=retries))
```

Or maybe this makes conan more resilient to network failures for most cases, which is a great thing? In this case, I am good with using also ``general.retry`` to configure this, maybe it doesn't deserve a dedicated config, as sounds like too fine grained and requiring too much explanation. 

Just trying to foresee any possible problems and understand the change, not saying yes or not.

Reply:
according to the https://github.com/kennethreitz/requests/issues/4506, this also happens in AWS clouds. given the fact Azure and AWS are two most popular and wide-spread cloud providers, I think it should be good to enable this, as we make conan more robust. as we haven't any reports, it means retries will never happen in most cases, right? so change will be safe in this case.
detecting Azure specifically in conan requires additional code, and I doubt it's useful to add it, as it adds unnecessary complexity: https://gallery.technet.microsoft.com/scriptcenter/Detect-Windows-Azure-aed06d51 (there are also variants to rely on IP ranges, but I think IPs may change over the time).
I also prefer to use just `general.retry`, as it fits Occam's razor, if we already have config for retires, why would we need to have one more config for retries?

Reply:
Sounds good to me! Thanks!